File: Week_02/G20190343020222/n-ary-tree-preorder-traversal.js
var preorder = function(root) {
    let stack = [root];
    const result = [];
    let current;
    while (stack.length > 0) {
        current = stack.shift();
        
        if (current) {
            result.push(current.val);
            stack = current.children.concat(stack);
        }
    }
    return result;
};

File: Week_02/G20190343020042/n-ary-tree-preorder-traversal.cpp
class Solution {
public:
    void pre(Node* root,vector<int>& t)
    {
        if(!root)
            return;
        t.push_back(root->val);
        for(auto p:root->children)
            pre(p,t);
    }
    vector<int> preorder(Node* root) {
        vector<int> result;
        pre(root,result);
        return result;
        
    }
};